File: doc/tips/distribution/amazon/dynamodb/python/cities_create_table.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
# --------------------------------------------------------------------
#from __future__ import print_function # Python 2/3 compatibility
import sys
import boto3

import sys
#

sys.stderr.write ("*** 開始 ***\n")

dynamodb = boto3.resource('dynamodb', region_name='us-west-2', endpoint_url="http://localhost:8000")



table = dynamodb.create_table(
    TableName='cities',
    KeySchema=[
        {
            'AttributeName': 'key',
            'KeyType': 'HASH'  #Partition key
        },
	{
            'AttributeName': 'name',
            'KeyType': 'RANGE'  #Sort key
        }
    ],
    AttributeDefinitions=[
        {
            'AttributeName': 'key',
            'AttributeType': 'S'
        },
        {
            'AttributeName': 'name',
            'AttributeType': 'S'
        },
    ],
    ProvisionedThroughput={
        'ReadCapacityUnits': 10,
        'WriteCapacityUnits': 10
    }
)

print("Table status:", table.table_status)
sys.stderr.write ("*** 終了 ***\n")
# --------------------------------------------------------------------

File: doc/tips/distribution/amazon/dynamodb/php_local/cities_create_table.php
#! /usr/bin/php
<?php
require 'vendor/autoload.php';

date_default_timezone_set('UTC');

use Aws\DynamoDb\Exception\DynamoDbException;

$sdk = new Aws\Sdk([
    'endpoint'   => 'http://localhost:8000',
    'region'   => 'us-west-2',
    'version'  => 'latest'
]);

$dynamodb = $sdk->createDynamoDb();

$params = [
    'TableName' => 'cities',
    'KeySchema' => [
        [
            'AttributeName' => 'key',
            'KeyType' => 'HASH'  //Partition key
        ],
        [
            'AttributeName' => 'name',
            'KeyType' => 'RANGE'  //Sort key
        ]
    ],
    'AttributeDefinitions' => [
        [
            'AttributeName' => 'key',
            'AttributeType' => 'S'
        ],
        [
            'AttributeName' => 'name',
            'AttributeType' => 'S'
        ],

    ],
    'ProvisionedThroughput' => [
        'ReadCapacityUnits' => 10,
        'WriteCapacityUnits' => 10
    ]
];

try {
    $result = $dynamodb->createTable($params);
    echo 'Created table.  Status: ' . 
        $result['TableDescription']['TableStatus'] ."\n";

} catch (DynamoDbException $e) {
    echo "Unable to create table:\n";
    echo $e->getMessage() . "\n";
}

?>
